### ISSUE TITLE
Issues from Android and iOS appear differently in the Sentry UI from a Flutter app

### ISSUE THREAD

**Issue by SoftMemes** [getsentry/sentry-dart#1983]
### Platform

Flutter Mobile

### Obfuscation

Enabled

### Debug Info

Enabled

### Doctor

[✓] Flutter (Channel stable, 3.16.9, on Ubuntu 23.10 6.5.0-27-generic, locale en_GB.UTF-8)
    • Flutter version 3.16.9 on channel stable at /home/freed/opt/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 41456452f2 (3 months ago), 2024-01-25 10:06:23 -0800
    • Engine revision f40e976bed
    • Dart version 3.2.6
    • DevTools version 2.28.5

### Version

7.16.1

### Steps to Reproduce

I have a Flutter project and build for Android and iOS. I am using the dart sentry plugin to upload debug symbols and sources, and use the Sentry CLI directly to create releases in Sentry. Further I'm using the Sentry CLI to upload pro-guard files for Android and explicity specify the UUID when initializing Sentry.

I've added my package name in `inAppIncludes` in the app.

I have also set up the Github integration and code mappings to point to the relevant directories.

The Android and iOS versions are built separately as part of CI. I use an automatically generated unique build number for each (time based) and infer the version from git tags, resulting in release identifiers liks `myapp-android@1.2.3-something+40001` and `myapp-ios@1.2.3-something+40041`. These are distinct releases/packages in Sentry. This all works and they show up as expected.

### Expected Result

- Android and iOS issues are grouped together.
- Source context is available for exceptions from my code
- In-app frames are detected and flagged
- Links to GitHub work on stack trace frames

### Actual Result

- Android and iOS issues show up separately, not grouped
- Source context for stacktraces _are_ showing up in Android issues
- Android stack frames are _not_ flagged as in-app
- Android issues do not have a link to Github
- Source context for stacktraces _are not_ visible for iOS issues
- iOS issues do have stack frames linked correctly to Github
- iOS issues have the stack frames correctly detected as in-app

Further, the paths reported are different between the two cases (which may explain some of the other issues/differences)
- For Android, the file path reported when hovering over the filename in the web UI is an absolute path to the file as it existed when the project was built (in this case a path starting with /home/circleci/project
- For iOS, the file path is prefixed with my package name (package:mypackage/somedir/file.dart). This path does not include "lib".

### Are you willing to submit a PR?

Yes

**Comment by buenaflor** [comment]
Hey can you send the links to these events to me at giancarlo.buenaflor@sentry.io so we can take a closer look

**Comment by buenaflor** [comment]
Could you also provide a reproducible sample app? It would greatly help in debuggin this

**Comment by denrase** [comment]
So i tried to reproduce this using the TestFlight iOS example, and an Android release build. For the latter, the plugin was also executed so new debug symbols were uploaded. I also set the build number to match iOS. At first this did

## List of Reported Issues/Problems

> Android and iOS issues show up separately, not grouped

Yes. The issues were correctly grouped for iOS (release) and Android (release) build. This was only not working when one platform did not have `--split-debug-info` and '--obfuscate' enabled and the other did.
 
| iOS (Release) | Android (Release) |
| --------------|------------------|
| <img width="1149" alt="Bildschirmfoto 2024-05-14 um 17 59 57" src="https://github.com/getsentry/sentry-dart/assets/3984453/4febea1f-33af-4449-ad88-13ca26ad35df">  | <img width="989" alt="Bildschirmfoto 2024-05-14 um 16 13 47" src="https://github.com/getsentry/sentry-dart/assets/3984453/ab5d498f-0b94-4224-b091-85d1dd6c0279"> |

> Source context for stacktraces are showing up in Android issues

Yes, as seen above.

> Android stack frames are not flagged as in-app

Yes, only in iOS.

| Not in-app |
|------------|
|  <img width="1819" alt="Bildschirmfoto 2024-05-14 um 17 54 44" src="https://github.com/getsentry/sentry-dart/assets/3984453/086da1cc-13a9-445f-a6c7-cc94c9c55c7a">  |   

> Android issues do not have a link to Github

Yes, see above.

> Source context for stacktraces are not visible for iOS issues

No, this works with `--split-debug-info` and '--obfuscate' flags. See screenshots above.

> iOS issues do have stack frames linked correctly to Github

No, this works with `--split-debug-info` and '--obfuscate' flags. See screenshots above.

> iOS issues have the stack frames correctly detected as in-app

Yes, see screenshots above.

## sentry.io Issues

[iOS (Release)](https://sentry-sdks.sentry.io/issues/5349252995/events/0690a879135742fc98aa2e744dadf714/?project=5428562&query=is%3Aunresolved+issue.priority%3A%5Bhigh%2C+medium%5D&referrer=next-event&statsPeriod=14d&stream_index=0)
[Android (Release)](https://sentry-sdks.sentry.io/issues/5349252995/events/f436ef5980b9409cbc73bc004d645dea/?project=5428562&query=is%3Aunresolved+issue.priority%3A%5Bhigh%2C+medium%5D&referrer=next-event&statsPeriod=14d&stream_index=0)

**Comment by denrase** [comment]
Still need to verify if we correctly setup the TestFlight release build.

**Comment by denrase** [comment]
Updated the comments above. It looks like everything works, except detecting in-app frames on Android and linking to GH.

**Comment by denrase** [comment]
@buenaflor Looking through this, i think we need to set the in-app flag on the backend. On the client, we do not have the package if the app is obfuscated, which is how in-app frames are determined.

| Default | Obfuscated |
| ------- | ----------- |
|  <img width="410" alt="Bildschirmfoto 2024-06-03 um 14 43 12" src="https://github.com/getsentry/sentry-dart/assets/3984453/7e98923a-45ff-4b5b-804a-3ff49425e144"> |  <img width="396" alt="Bildschirmfoto 2024-06-03 um 14 44 46" src="https://github.com/getsentry/sentry-dart/assets/3984453/2a177d12-3b70-4dc4-a6d2-5383289a45b4"> |

**Comment by buenaflor** [comment]
@Swatinem hey, you got a clue for two identical stacktraces (iOS & Android) where one has every frame marked as inApp (iOS) and the other does not have any marked(Android)? I guess the stacktrace rules are affecting it?

**Comment by Swatinem** [comment]
Yes, the stacktrace rules have an effect there. There is also `in_app` normalization, which defaults a `None` value to true/false depending on presence of either flag anywhere else in the stack trace. So its possible that iOS (either through the SDK, or a builtin enhancer rule) sets that flag on one system frame, and then other frames are being defaulted.
I don’t have enough context here to be of more help.

**Comment by buenaflor** [comment]
@krystofwoldrich in RN triggering the same error on iOS and Android will result being grouped correct? Do you do anything specific there to make that happen?

**Comment by krystofwoldrich** [comment]
If RN App throws error in JS `throw new Error('My first error')` this will result in one issue in Sentry for both iOS and Android, because the `in_app` JS frames will be the same.

**Comment by buenaflor** [comment]
yeah makes sense, I suppose we also want the same behaviour for Flutter

**Comment by buenaflor** [comment]
Improvements are live:
- stacktraces on both ios and android are now properly set inApp
- iOS and Android events of the same error are now grouped together

**Comment by buenaflor** [comment]
I will close this issue and create a new issue regarding the Github stacktrace links
